keras/engine/training.py

```diff
@@ -962,10 +962,21 @@ def _make_predict_function(self):
                                                name='predict_function',
                                                **kwargs)
 
+    def _check_num_samples(self, ins, batch_size=None, steps=None, steps_name='steps'):
+        if steps is not None:
+            num_samples = steps
```

Comment:
Perhaps something should be renamed in this case or documentation modified a bit for everything to be consistent?

I wasn't sure if you're saying I should make it `step*batch_size` here or if you're saying that's also inaccurate and we need to do something else.